src/system-update-generator/system-update-generator.c

```diff
@@ -47,9 +48,41 @@ static int generate_symlink(void) {
         if (symlink(SYSTEM_DATA_UNIT_PATH "/system-update.target", p) < 0)
                 return log_error_errno(errno, "Failed to create symlink %s: %m", p);
 
+        return 1;
+}
+
+static int parse_proc_cmdline_item(const char *key, const char *value, void *data) {
+        assert(key);
+
+        if (streq(key, "systemd.unit")) {
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
+
+                log_warning("Offline system update overriden by kernel command line systemd.unit= setting");
+
+        } else if (!value) {
+                const char *target;
+
+                target = runlevel_to_target(key);
+                if (target)
+                        log_warning("Offline system update overriden by runlevel \"%s\" on the kernel command line", key);
+        }
+
         return 0;
 }
 
+static void maybe_warn(void) {
+        /* Check if a run level is specified on the kernel command line. The
+         * command line has higher priority than any on-disk configuration, so
+         * it'll make any symlink we create moot.
+         */
+        int r;
+
+        r = proc_cmdline_parse(parse_proc_cmdline_item, NULL, 0);
+        if (r < 0)
+                log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
```

Comment:
nitpick: No need to have `r` here, and since the entire function is so small and only being used once, it might be easier to just fold it into `main()` below?